app/(main)/league/[leagueId]/entry/[entryId]/week/Week.test.tsx

```diff
@@ -153,22 +167,106 @@ describe('Week', () => {
     });
   });
 
-  test('should not display GlobalSpinner after loading data', async () => {
+  test('should display main content after data is loaded and hide GlobalSpinner', async () => {
     mockUseAuthContext.isSignedIn = true;
-    mockCreateWeeklyPicks.mockResolvedValue({});
+    (getCurrentUserEntries as jest.Mock).mockResolvedValue([
+      {
+        $id: '123',
+        name: 'Entry 1',
+        user: '123',
+        league: '123',
+        selectedTeams: [],
+        eliminated: false,
+      },
+    ]);
 
     render(
       <Week entry={entry} league={league} NFLTeams={NFLTeams} week={week} />,
     );
-    await waitFor(() => {
-      expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
-    });
+
+    // Wait for the main content to be displayed
+    await waitFor(
+      () => {
+        expect(screen.getByTestId('weekly-picks')).toBeInTheDocument();
+      },
+      { timeout: 5000 },
+    );
+
+    // Ensure the GlobalSpinner is not present when main content is loaded
+    expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
+  });
+
+  test('should not show previous weeks picks if the current week is not greater than 0', async () => {
```

Comment:
Based on your value of week, which is `1` and is greater than `0`, this test should not passing based on this description and your test logic.  Please feel free to correct me if I did not understand what this test it doing.

Reply:
@choir27 great catch! I meant to update the description to `'should not show previous weeks picks if there is no history of selected teams'`

✅ updated!